doc/radosgw/admin.rst

```diff
@@ -211,61 +211,104 @@ Options include:
+- ``--gen-secret`` generates a random S3 secret key.
 
-To create a key for a user, you must specify ``key create``. For a user, specify
-the user ID and the ``s3`` key type. To create a key for subuser, you must
-specify the subuser ID and the ``swift`` keytype. For example::
+An example how to create/add a specified S3 key pair for a user. ::
 
-	radosgw-admin key create --subuser=johndoe:swift --key-type=swift --gen-secret
+	radosgw-admin key create --uid=foo --access-key fooAccessKey --secret-key fooSecretKey
 
 .. code-block:: javascript
 
-  { "user_id": "johndoe",
+  { "user_id": "foo",
     "rados_uid": 0,
-    "display_name": "John Doe",
-    "email": "john@example.com",
+    "display_name": "foo",
+    "email": "foo@example.com",
+    "suspended": 0,
+    "keys": [
+      { "user": "foo",
+        "access_key": "fooAccessKey",
+        "secret_key": "fooSecretKey"}],
+  }
+
+To generate random key pair, use the ``--gen-access-key`` option instead. ::
+
+	radosgw-admin key create --uid=foo --gen-access-key
+ 
+Note that if you create different S3 key pairs for a user, the user will have multple keys.
+
+To add a swift secret for a subuser. ::
+
+	radosgw-admin key create --subuser=foo:bar --key-type=swift --secret barSecret
+
+.. code-block:: javascript
+
+  { "user_id": "foo",
+    "rados_uid": 0,
+    "display_name": "foo",
+    "email": "foo@example.com",
     "suspended": 0,
     "subusers": [
-       { "id": "johndoe:swift",
+       { "id": "foo:bar",
          "permissions": "full-control"}],
-    "keys": [
-      { "user": "johndoe",
-        "access_key": "QFAMEDSJP5DEKJO0DDXY",
-        "secret_key": "iaSFLDVvDdQt6lkNzHyW4fPLZugBAI1g17LO0+87"}],
     "swift_keys": [
-      { "user": "johndoe:swift",
-        "secret_key": "E9T2rUZNu2gxUjcwUBO8n\/Ev4KX6\/GprEuH4qhu1"}]}
+      { "user": "foo:bar",
+        "secret_key": "asfghjghghmgm"}]}
 
+You may also specify the secret.::
 
+	radosgw-admin key create --subuser=foo:bar --gen-access-key
 
-Add / Remove Access Keys
-------------------------
+Note that a subuser can only have one swift secret.
 
-Users and subusers must have access keys to use the S3 and Swift
-interfaces. When you create a user or subuser and you do not specify 
-an access key and secret, the key and secret get generated automatically. 
-You may create a key and either specify or generate the access key and/or
-secret. You may also remove an access key and secret. Options include:
+In the least radosgw version, subuser can also operates S3 APIs. To create a S3
+key pair for a subuser. ::
 
+	radosgw-admin key create --subuser=foo:bar --key-type=s3 --gen-access-key
 
-- ``--secret=<key>`` specifies a secret key (e.g,. manually generated).
-- ``--gen-access-key`` generates random access key (for S3 user by default).
-- ``--gen-secret`` generates a random secret key.
-- ``--key-type=<type>`` specifies a key type. The options are: swift, s3
+or sepficy the key pair. ::
```

Comment:
the `--gen-access-key` flag was already described above, so i think it's redundant to give both examples (`--gen-access-key` and `--access-key`/`--secret-key`) for subusers also. let's just keep this one example with `--access-key barAccessKey --secret-key barSecretKey` so it matches the json output below